docs/working-with-files.md

```diff
@@ -228,29 +228,43 @@ In general, you should not need to manually copy files, because Nextflow will au
 
 ## Remote files
 
-Nextflow can work with many kinds of remote files and objects using the same interface as for local files. The following protocols are supported:
+Nextflow works with many types of remote files and objects using the same interface as for local files. The following protocols are supported:
 
-- HTTP(S) / FTP (`http://`, `https://`, `ftp://`)
+- HTTP(S)/FTP (`http://`, `https://`, `ftp://`)
 - Amazon S3 (`s3://`)
 - Azure Blob Storage (`az://`)
 - Google Cloud Storage (`gs://`)
 
-To reference a remote file, simple specify the URL when opening the file:
+To reference a remote file, simply specify the URL when opening the file:
 
 ```nextflow
 pdb = file('http://files.rcsb.org/header/5FID.pdb')
 ```
 
-You can then access it as a local file as described previously:
+It can then be used in the same way as a local file:
 
 ```nextflow
 println pdb.text
 ```
 
 :::{note}
-Not all operations are supported for all protocols. In particular, writing and directory listing are not supported for HTTP(S) and FTP paths.
+Not all operations are supported for all protocols. For example, writing and directory listing is not supported for HTTP(S) and FTP paths.
 :::
 
 :::{note}
-Additional configuration may be required to work with cloud object storage (e.g. to authenticate with a private bucket). Refer to the respective page for each cloud storage provider for more information.
+Additional configuration may be necessary for cloud object storage, such as authenticating with a private bucket. See the documentation for each cloud storage provider for further details.
+:::
+
+### Remote file staging
+
+When a remote file is passed as an input to a process, Nextflow stages the file into the work directory using an appropriate Java SDK.
+
+Remote files are staged in a subdirectory of the work directory with the form `stage-<session-id>/<hash>/<filename>`, where `<hash>` is determined by the remote file path. If multiple tasks request the same remote file, the file will be downloaded once and reused by each task. These files can be reused by resumed runs with the same session ID.
```

Comment:
It would be better to better this kind of file transfer happens every time the origin or the destination of the file system is different from the workflow work directory. 

For example, the input file is in the local computer or it's a http remote file AND the pipeline uses S3 bucket as work dir, then nextflow needs to copy into S3. Same logic when it needs to copy the output files.  

For the same reason it's important to advice to keep the input and outputs in the same storage system e.g. S3 or shared file system. 


Minor: it would be preferable use "copy" or download input files and upload output files instead instead of "stage" that's too slang tech term.